executor/aggfuncs/func_avg.go

```diff
@@ -205,3 +267,54 @@ func (e *avgPartial4Float64) UpdatePartialResult(sctx sessionctx.Context, rowsIn
 	}
 	return nil
 }
+
+type partialResult4AvgDistinctFloat64 struct {
+	partialResult4AvgFloat64
+	exists map[float64]bool
```

Comment:
ditto

Reply:
`map[types.MyDecimal]bool` is much convenient to use and will not introduce much memory overhead.

Reply:
Here is an experiment:
- `map[int]bool` with 10M elements: 315M 
- `map[int]struct{}` with 10M elements: 286M

## 1. memory usage of `map[int]bool`
```go
package main

import (
        "fmt"
        "runtime"
)

func main() {
        numElements := 10000000
        var m1, m2 runtime.MemStats

        runtime.ReadMemStats(&m1)
        a := make(map[int]bool)
        for i := 0; i < numElements; i++ {
                a[i] = true
        }
        runtime.ReadMemStats(&m2)
        fmt.Println("Bytes allocated for A:", m2.Alloc-m1.Alloc)
}
```

The result is:
```sh
[jianzhang.zj:/tmp]
➜ go build a.go

[jianzhang.zj:/tmp]
➜ ./a
Bytes allocated for A: 315457888
```

## 2. memory usage of `map[int]struct{}`
```go
package main

import (
        "fmt"
        "runtime"
)

func main() {
        numElements := 10000000
        var m1, m2 runtime.MemStats

        runtime.ReadMemStats(&m1)
        b := make(map[int]struct{})
        for i := 0; i < numElements; i++ {
                b[i] = struct{}{}
        }
        runtime.ReadMemStats(&m2)
        fmt.Println("Bytes allocated for B:", m2.Alloc-m1.Alloc)
}
```

the result is:
```sh
[jianzhang.zj:/tmp]
➜ go build b.go

[jianzhang.zj:/tmp]
➜ ./b
Bytes allocated for B: 286228080
```